File: HanYeojin/Assignment2/yeojin/cafe-server/src/main/java/com/yeojin/cafe/service/MenuService.java
package com.yeojin.cafe.service;

import com.yeojin.cafe.domain.Menu;
import com.yeojin.cafe.dto.MenuResponseDto;
import com.yeojin.cafe.repository.MenuRepository;
import org.springframework.stereotype.Service;

import java.util.*;
import java.util.stream.Collectors;

@Service
public class MenuService {
    private final MenuRepository menuRepository;

    // 생성자 주입 = @RequiredArgsConstructor
    public MenuService(MenuRepository menuRepository) {
        this.menuRepository = menuRepository;
    }

    // 전체 메뉴 조회
    public List<MenuResponseDto> getAllMenus() {
        List<Menu> menus = menuRepository.findAllMenu();

        // Entity -> DTO 변환
        return menus.stream()
                .map(menu -> new MenuResponseDto(
                        menu.getId(),
                        menu.getName(),
                        menu.getPrice()
                ))
                .collect(Collectors.toList());
    }

}

File: assignment2/MinSeungho/cafe/src/main/kotlin/com/seungho/cafe/service/MenuService.kt
package com.seungho.cafe.service

import com.seungho.cafe.dto.MenuResponse
import com.seungho.cafe.repository.MenuRepository
import org.springframework.stereotype.Service

@Service
class MenuService (
    private val menuRepository: MenuRepository
){
    fun getAllMenus(): List<MenuResponse> {
        val menus = menuRepository.findAll()

        return menus.map {
                menu -> MenuResponse.from(menu)
        }
    }
}